eos/scattering/parametric-gmkprdey2011.cc

```diff
@@ -0,0 +1,391 @@
+                        UsedParameter(p[_par_name("S0", "D")],      *this)
+        },
+        _params_P1 {    UsedParameter(p[_par_name("P1", "B", 0)],   *this),
+                        UsedParameter(p[_par_name("P1", "B", 1)],   *this),
+                        UsedParameter(p[_par_name("P1", "lam", 1)], *this),
+                        UsedParameter(p[_par_name("P1", "lam", 2)], *this)
+        },
+        _params_D0 {    UsedParameter(p[_par_name("D0", "B", 0)],   *this),
+                        UsedParameter(p[_par_name("D0", "B", 1)],   *this),
+                        UsedParameter(p[_par_name("D0", "Bh", 1)],  *this)
+        },
+        _mPi(UsedParameter(p["mass::pi^+@GMKPRDEY2011"], *this)),
+        _mK(UsedParameter(p["mass::K_u@GMKPRDEY2011"], *this)),
+        _mEta(UsedParameter(p["mass::eta@GMKPRDEY2011"], *this)),
+        _mRho(UsedParameter(p["mass::rho^0@GMKPRDEY2011"], *this)),
+        _mF2(UsedParameter(p["mass::f_2@GMKPRDEY2011"], *this)),
+        _sM_S0(UsedParameter(p[_par_name("S0", "sM")],  *this)),
+        _s0_P1(UsedParameter(p[_par_name("P1", "s0")],  *this)),
+        _s0_D0(UsedParameter(p[_par_name("D0", "s0")],  *this)),
+        _sh_D0(UsedParameter(p[_par_name("D0", "sh")],  *this)),
+        _cont_pow_S0(UsedParameter(p[_par_name("S0", "n")],  *this)),
+        _cont_pow_P1(UsedParameter(p[_par_name("P1", "n")],  *this)),
+        _cont_pow_D0(UsedParameter(p[_par_name("D0", "n")],  *this)),
+        _intervals_P1({4 * _mPi * _mPi , 0.5 , 1.0 , 2.0 }),
+        _intervals_D0({4 * _mPi * _mPi , 0.7, 1.1, 1.45, 2.0 }),
+        _f_phase_P1(std::bind(&GMKPRDEY2011ScatteringAmplitudes::_phase_P1, this, std::placeholders::_1)),
+        _f_phase_D0(std::bind(&GMKPRDEY2011ScatteringAmplitudes::_phase_D0, this, std::placeholders::_1)),
+        _omnes_P1(_intervals_P1, _f_phase_P1, 0.0),
+        _omnes_D0(_intervals_D0, _f_phase_D0, 0.0)
+    {
+    }
+
+    GMKPRDEY2011ScatteringAmplitudes::~GMKPRDEY2011ScatteringAmplitudes() = default;
+
+    ScatteringAmplitudes<PPToPP> *
+    GMKPRDEY2011ScatteringAmplitudes::make(const Parameters & parameters, const Options & options)
+    {
+        return new GMKPRDEY2011ScatteringAmplitudes(parameters, options);
+    }
+
+    QualifiedName
+    GMKPRDEY2011ScatteringAmplitudes::_par_name(const std::string & partial_wave, const std::string & par_name, unsigned idx) const
+    {
+        return QualifiedName(stringify(PiPiToPiPi::label) + "::" + partial_wave + "_" + par_name + "_" + stringify(idx) + "@GMKPRDEY2011");
+    }
+
+    QualifiedName
+    GMKPRDEY2011ScatteringAmplitudes::_par_name(const std::string & partial_wave, const std::string & par_name) const
+    {
+        return QualifiedName(stringify(PiPiToPiPi::label) + "::" + partial_wave + "_" + par_name + "@GMKPRDEY2011");
+    }
+
+    complex<double> GMKPRDEY2011ScatteringAmplitudes::scattering_amplitude(const double & s, const unsigned & l, const std::string & i) const
```

Comment:
Can we make i an integer or at least an enum?

Reply:
Please use a quantum number for this (i.e. an enum).

Reply:
Implemented IsospinRepresentation for I = 0, 1, 2, 1/2 & 3/2.